ata/atapicdr, scsi/scsicd: Set CDDA image on device reset

987123879113 opened the pull request:
This fixes the recently broken CDDA audio for CD-ROM drives. nscsi already had set_cdrom called on device_reset so no change required there. Tested on Konami GV, Konami Firebeat (ppp series), and Konami System 573 (early DDR games before 3rd mix).

The reason these broke is the DVD commit (https://github.com/mamedev/mame/commit/28104cdbdfc39b0ced6411381ffb074772dce345) removed `t10mmc::SetDevice` which was called when an image was changed and was responsible for calling `m_cdda->set_cdrom(...)`.

@galibert Please review and fix any remaining issues with CDDA before this month's release.

einstein95 commented:
Pinging @angelosa given their comment on the regressions caused by that commit which included the Konami GV

angelosa commented:
There's not much to add here: it fixes t10mmc specific issues but won't fix the rest (PCE CD, Saturn, PS1 etc. etc.)

987123879113 commented:
Added the fix for Konami M2 CDs + removed old code that is unused now + removed evilngte since it's just a old incomplete dupe of hellngt (the European version).
